src/indexer.rs

```diff
@@ -571,21 +617,31 @@ impl Indexer for AffectedContractIndexer {
         mut client: v2::Client,
         _ctx: &'a (),
         fbi: FinalizedBlockInfo,
-    ) -> QueryResult<Self::Data> {
+    ) -> OnFinalizationResult<Self::Data> {
         let bi = client.get_block_info(fbi.height).await?.response;
         if bi.transaction_count != 0 {
             let summary = client
                 .get_block_transaction_events(fbi.height)
-                .await?
+                .await
+                .map_err(OnFinalizationError::from)?
                 .response
+                .map_err(OnFinalizationError::from)
                 .try_filter_map(|summary| async move {
                     let Some(info) = update_info(summary) else {
                         return Ok(None);
                     };
-                    let affected_addresses = info.execution_tree.affected_addresses();
+
+                    let v2::Upward::Known(ref execution_tree) = info.execution_tree else {
+                        return Err(OnFinalizationError::Unknown(
+                            "contractUpdateInfo.execution_tree".to_string(),
+                        ));
+                    };
+
+                    let affected_addresses = execution_tree.affected_addresses();
                     let v2::Upward::Known(affected_addresses) = affected_addresses else {
                         return Ok(Some(v2::Upward::Unknown));
```

Comment:
Note: Currently the indexer returns an error if the `contractUpdateInfo.execution_tree` is unkown (which happens if any smart contract version is unkown that the tx passes through).

In the `execution_tree.affected_addresses()` we return `Ok(Some(v2::Upward::Unknown))` and allow the user to handle that case. This can happen if any of the `events/executionTraces` are unknown to the SDK.

The current way distinguish both cases, but  we could also argue that both cases should or should not return an error or should return an `Unkown`.